netty/src/main/java/io/grpc/netty/NettyServerTransport.java

```diff
@@ -164,12 +170,35 @@ private void notifyTerminated(Throwable t) {
     }
   }
 
+  @Override
+  @Nullable
+  public TransportTracer.Stats getTransportStats() {
+    if (channel.eventLoop().inEventLoop()) {
+      // This is necessary, otherwise we will block forever on the promise in the else case
+      return transportTracer.getStats();
+    }
+    final Promise<TransportTracer.Stats> promise = channel.eventLoop().newPromise();
+    channel.eventLoop().execute(new Runnable() {
```

Comment:
Probably want to use `submit()` here.  If there is an exception, the promise may not be resolved.

Reply:
Done.